packages/contracts/contracts/registry/Registry.sol

```diff
@@ -4,36 +4,49 @@
 
 pragma solidity 0.8.10;
 
-import "../core/DAO.sol";
+import "../core/component/Component.sol";
+
+/// @title Register a DAO
+/// @author Aragon Association - 2022
+/// @notice This contract allows to register a DAO by its address
+contract Registry is Component {
+    bytes32 public constant REGISTER_DAO_ROLE = keccak256("REGISTER_DAO_ROLE");
+
+    mapping(IDAO => bool) public daos;
+
+    bytes4 internal constant REGISTRY_INTERFACE_ID = this.register.selector;
 
-/// @title Register your unique DAO name
-/// @author Samuel Furter - Aragon Association - 2021
-/// @notice This contract provides the possiblity to register a DAO by a unique name.
-contract Registry {
     /// @notice Thrown if the DAO name is already registered
-    /// @param name The DAO name requested for registration
-    error RegistryNameAlreadyUsed(string name);
+    /// @param dao The address requested for registration
+    error DAOAlreadyRegistered(IDAO dao);
 
     /// @notice Emitted if a new DAO is registered
     /// @param dao The address of the DAO contract
     /// @param creator The address of the creator
-    /// @param name The name of the DAO
-    event NewDAORegistered(DAO indexed dao, address indexed creator, string name);
+    event NewDAORegistered(IDAO indexed dao, address indexed creator, string name);
+
+    /// @notice Initializes the component
+    /// @dev This is required for the UUPS upgradability pattern
+    /// @param _dao The IDAO interface of the associated DAO
+    function initialize(IDAO _dao) external initializer {
+        _registerStandard(REGISTRY_INTERFACE_ID);
+
+        __Component_init(_dao);
+    }
 
-    mapping(string => bool) public daos;
+    // This function is intended to be called by the `DAOFactory`
 
-    /// @notice Registers a DAO by his name
+    /// @notice Registers a DAO by its address
     /// @dev A name is unique within the Aragon DAO framework and can get stored here
-    /// @param name The name of the DAO
     /// @param dao The address of the DAO contract
     function register(
-        string calldata name,
-        DAO dao,
-        address creator
-    ) external {
-        if (daos[name] != false) revert RegistryNameAlreadyUsed({name: name});
+        IDAO dao,
+        address creator,
+        string calldata name
+    ) external auth(REGISTER_DAO_ROLE) {
```

Comment:
Wouldn't this go against the possibility of anyone being able to register their DAO?

Reply:
No, we can grant permission to `ANY_ADDR` explicitly (see below) https://github.com/aragon/core/blob/f1ba89e79e39079dfbc84584c60494aab21aa035/packages/contracts/contracts/core/acl/ACL.sol#L54 
which is checked here

https://github.com/aragon/core/blob/f1ba89e79e39079dfbc84584c60494aab21aa035/packages/contracts/contracts/core/acl/ACL.sol#L167-L177

In this case, anyone can use the `register` method.
In practice and as discussed with @githubdoramon, we probably want to give `REGISTER_DAO_ROLE ` to the `DAOFactory` that @Rekard0 is working on and potentially to a self-registration contract [as discussed here](https://aragonassociation.atlassian.net/browse/APP-407) that could even natively support other DAO frameworks (e.g. Zodiac, Moloch, Colony etc.) to make it easy for the user. However, we don't need to worry about self-registration now with this design.